### DISCUSSION TITLE
[VOTE] Gradle script style

### DISCUSSION THREAD

**Discussion by zyxkad** [IntelligenceModding/AdvancedPeripherals#627] [General]
I'm currently trying to make a fabric port. (just trying, not promising anything)
However, when I write gradle scripts, I found there are a lot single quotation mark strings `'xxx'`, as well as a lot double quotation mark strings `"xxx"`.
Then the script drive me crazy because puzzled format styles.

I make a few votes below, you can react on the emojis to choice which way you prefer.
However if I did not explain the options well, or if there are more choices, you can reply on the comments, or write a comment for new suggestion

**Comment by SirEndii** [comment]
Yeah that's my fault
I am not sure if there are any general conventions for groovy gradle build scripts, but I would stick with double quotation marks

**Comment by zyxkad** [comment]
Ok, so let's make some decisions on the styles:

1. SirEndii said always use double quotations, that's perfect imo.

But let's still vote for it, if you thing we should use double quotations for strings, react 👍 
If you think we should use single quotations for strings, react ❤️

**Comment by zyxkad** [comment]
2. When we write links, should we have trailing slash?
If we should have, e.g. (`https://mvn.intelligence-modding.de/Intelligence/`) give an emoji 👍 
If we should not have, e.g. (`https://mvn.intelligence-modding.de/Intelligence`) give an emoji ❤️

**Comment by zyxkad** [comment]
3. Sometimes gradle let us to ignore brackets.

If you think we should ignore brackets as much as possible, click 👍 
```
        content {
            includeGroup "net.minecraftforge"
        }
```

If you think we should not ignore brackets, click ❤️ 
```
        content {
            includeGroup("net.minecraftforge")
        }
```

**Comment by zyxkad** [comment]
4. And gradle allow us to ignore `=` in some cases.

If you think we should ignore equal signs as much as possible, click 👍 
```
maven {
    name "MinecraftForge"
    url "https://maven.minecraftforge.net"
}
```

If you think we should not ignore equals, click ❤️ 
```
maven {
    name = "MinecraftForge"
    url = "https://maven.minecraftforge.net"
}
```

**Comment by zyxkad** [comment]
5. when assigning value or invoke methods, gradle also let us to not nest many brackets, but use dot `.` instead

If you think we should always use brackets, click 👍  (imo that's cleaner than dots)
```
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(java_version)
    }
    withSourcesJar()
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}
jar {
    finalizedBy('reobfJar')
}
```
If you think we should use dots for single statement, click ❤️ 
```
java {
    toolchain.languageVersion = JavaLanguageVersion.of(java_version)
    withSourcesJar()
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}
jar.finalizedBy('reobfJar')
```
If you think we should always and only use the dots outside the brackets, click 🚀 
```
java.toolchain.languageVersion = JavaLanguageVersion.of(java_version)
java {
    withSourcesJar()
    sourceCompatibility = JavaVersion.VERSION_17
    targetCompatibility = JavaVersion.VERSION_17
}
jar.finalizedBy('reobfJar')
```

**Comment by zyxkad** [comment]
6. When using gradle properties, we have three choices

- `something = mod_version + "xxx"`, react 👍 
- `something = "$mod_version" + "xxx"`, react ❤️ 
- `something = "${mod_version}xxx"`, react 🚀

**Comment by SirEndii** [comment]
for some reason I didn't saw the votes
I now reacted to your questions

**Reply by zyxkad** [reply]
I wrote this vote when I had almost zero understanding on gradle. 😅 
some options need better explanation, but I think our project is not as large right now, so this is kind low priority.
